res/controllers/Pioneer-DDJ-SB2-scripts.js

```diff
@@ -0,0 +1,1187 @@
+PioneerDDJSB2.twinkleVumeterAutodjOn = true;
+
+// If true, by release browser knob jump forward to "position". 
+PioneerDDJSB2.jumpPreviewEnabled = true;
+PioneerDDJSB2.jumpPreviewPosition = 0.5;
+
+
+///////////////////////////////////////////////////////////////
+//               INIT, SHUTDOWN & GLOBAL HELPER              //
+///////////////////////////////////////////////////////////////
+PioneerDDJSB2.longButtonPress = false;
+PioneerDDJSB2.speedRateToNormalTimer = new Array(4);
+
+PioneerDDJSB2.init = function(id) {
+    PioneerDDJSB2.scratchSettings = {
+        'alpha': 1.0 / 8,
+        'beta': 1.0 / 8 / 32,
+        'jogResolution': 720,
+        'vinylSpeed': 33 + 1 / 3,
+        'safeScratchTimeout': 20
+    };
+
+    PioneerDDJSB2.channelGroups = {
+        '[Channel1]': 0x00,
+        '[Channel2]': 0x01,
+        '[Channel3]': 0x02,
+        '[Channel4]': 0x03
+    };
+
+    PioneerDDJSB2.samplerGroups = {
+        '[Sampler1]': 0x00,
+        '[Sampler2]': 0x01,
+        '[Sampler3]': 0x02,
+        '[Sampler4]': 0x03
+    };
+
+    PioneerDDJSB2.fxGroups = {
+        '[EffectRack1_EffectUnit1]': 0x00,
+        '[EffectRack1_EffectUnit2]': 0x01
+    };
+
+    PioneerDDJSB2.fxControls = {
+        'group_[Channel1]_enable': 0x00,
+        'group_[Channel3]_enable': 0x00,
+        'group_[Headphone]_enable': 0x01,
+        'group_[Master]_enable': 0x01,
+        'group_[Channel2]_enable': 0x02,
+        'group_[Channel4]_enable': 0x02
+    };
+
+    PioneerDDJSB2.shiftPressed = false;
+
+    PioneerDDJSB2.chFaderStart = [
+        null,
+        null
+    ];
+
+    PioneerDDJSB2.fxButtonPressed = [
+        [false, false, false],
+        [false, false, false]
+    ];
+
+    // used for soft takeover workaround
+    PioneerDDJSB2.fxParamsActiveValues = [
+        [0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0]
+    ];
+
+    PioneerDDJSB2.scratchMode = [false, false, false, false];
+
+    PioneerDDJSB2.ledGroups = {
+        'hotCue': 0x00,
+        'autoLoop': 0x10,
+        'manualLoop': 0x20,
+        'sampler': 0x30
+    };
+
+    PioneerDDJSB2.nonPadLeds = {
+        'headphoneCue': 0x54,
+        'shiftHeadphoneCue': 0x68,
+        'cue': 0x0C,
+        'shiftCue': 0x48,
+        'keyLock': 0x1A,
+        'shiftKeyLock': 0x60,
+        'play': 0x0B,
+        'shiftPlay': 0x47,
+        'vinyl': 0x17,
+        'shiftVinyl': 0x4E,
+        'sync': 0x58,
+        'shiftSync': 0x5C
+    };
+
+    PioneerDDJSB2.valueVuMeter = {
+        '[Channel1]_current': 0,
+        '[Channel2]_current': 0,
+        '[Channel3]_current': 0,
+        '[Channel4]_current': 0,
+        '[Channel1]_enabled': 1,
+        '[Channel2]_enabled': 1,
+        '[Channel3]_enabled': 1,
+        '[Channel4]_enabled': 1,
+
+    };
+
+    PioneerDDJSB2.loopIntervals = [1, 2, 4, 8, 16, 32, 64];
+
+    PioneerDDJSB2.looprollIntervals = [1 / 16, 1 / 8, 1 / 4, 1 / 2, 1, 2, 4, 8];
+
+    PioneerDDJSB2.setAllSoftTakeover(false);
```

Comment:
i have tested, some is functional correct, some not. Example Softtakeover: Working correct: pregain & volume. Not working: rate & crossfader. No idea what happens, its a todo for later.